opencv/src/OpenCV/Core/Types/Mat.hs

```diff
@@ -292,13 +296,14 @@ matFromFunc shape channels depth func =
 --------------------------------------------------------------------------------
 
 matCopyToM
-    :: (PrimMonad m)
+    :: (PrimMonad m, MonadError CvException m)
     => Mut (Mat ('S [dstHeight, dstWidth]) channels depth) (PrimState m) -- ^
     -> V2 Int32 -- ^
     -> Mat ('S [srcHeight, srcWidth]) channels depth -- ^
     -> Maybe (Mat ('S [srcHeight, srcWidth]) ('S 1) ('S Word8))
-    -> CvExceptT m ()
-matCopyToM dstM (V2 x y) src mbSrcMask = ExceptT $
+    -> m ()
+matCopyToM dstM (V2 x y) src mbSrcMask =
+    (>>= either throwError return) $
```

Comment:
Could we also generalize `handleCvException` like below so that we don't need these `(>>= either throwError return)` conversions?

```haskell
handleCvException
    :: (MonadError CvException m) 
    => IO a
    -> IO (Ptr (C CvCppException))
    -> IO (m a)
handleCvException okAct act = mask_ $ do
    exceptionPtr <- act
    if exceptionPtr /= nullPtr
      then do cppErr <- fromPtr (pure exceptionPtr)
              pure $ throwError $ BindingException cppErr
      else pure <$> okAct
```

Reply:
I tried a few things, but that may well fit!

Reply:
I actually just tried this but failed to get it past the type checker without the `(>>= either throwError return)`.

Reply:
Ah, then that may be why I left it like this. I do remember battling with this and ultimately surrendering and fixing up a few call sites.